workflow/rocoto/gfs_tasks.py

```diff
@@ -521,9 +525,10 @@ def _fcst_cycled(self):
             dep_dict = {'type': 'task', 'name': f'{self.cdump}ocnanalpost'}
             dependencies.append(rocoto.add_dependency(dep_dict))
 
-        if self.app_config.do_wave and self.cdump in self.app_config.wave_cdumps:
-            dep_dict = {'type': 'task', 'name': f'{self.cdump}waveprep'}
-            dependencies.append(rocoto.add_dependency(dep_dict))
+        # TODO: Cleanup for PR.
+        #if self.app_config.do_wave and self.cdump in self.app_config.wave_cdumps:
+        #    dep_dict = {'type': 'task', 'name': f'{self.cdump}waveprep'}
+        #    dependencies.append(rocoto.add_dependency(dep_dict))
```

Comment:
This dependency needs to be retained, but should no longer be grouped with the others that are compared against the OR with previous cycle exists. These need to always be satisfied, even for the half-cycle.

Reply:
Please see my lasted push. The result is the following XML block:

```
        <dependency>
                <or>
                        <and>
                                <taskdep task="gdassfcanl"/>
                        </and>
                        <not><cycleexistdep cycle_offset="-06:00:00"/></not>
                </or>
                <taskdep task="gdaswaveprep"/>
        </dependency>
```

Reply:
Needs to be one more `<and>` surrounding everything between the `<dependency>` tags.

Reply:
The last commit should resolve this. I will test the first half cycle and make sure this works.